File: src/main/java/com/andrew121410/mc/ccminecraftbot/packets/handle/pos/OnServerSpawnPositionPacket.java
package com.andrew121410.mc.ccminecraftbot.packets.handle.pos;

import com.andrew121410.mc.ccminecraftbot.CCBotMinecraft;
import com.andrew121410.mc.ccminecraftbot.packets.PacketHandler;
import com.andrew121410.mc.ccminecraftbot.player.CCPlayer;
import com.andrew121410.mc.ccminecraftbot.player.MovementManager;
import com.andrew121410.mc.ccminecraftbot.world.Location;
import com.github.steveice10.mc.protocol.packet.ingame.server.world.ServerSpawnPositionPacket;
import lombok.AllArgsConstructor;

@AllArgsConstructor
public class OnServerSpawnPositionPacket extends PacketHandler<ServerSpawnPositionPacket> {

    @Override
    public void handle(ServerSpawnPositionPacket packet, CCBotMinecraft ccBotMinecraft) {
        CCPlayer ccPlayer = ccBotMinecraft.getPlayer();
        ccPlayer.setCurrentLocation(Location.Companion.from(packet.getPosition()));
        ccPlayer.setSpawnPoint(Location.Companion.from(packet.getPosition()));
        ccPlayer.setMovementManager(new MovementManager(ccPlayer));
    }
}

File: src/main/kotlin/com/andrew121410/mc/ccminecraftbot/packets/handle/position/OnServerSpawnPositionPacket.kt
package com.andrew121410.mc.ccminecraftbot.packets.handle.position

import com.andrew121410.mc.ccminecraftbot.CCBotMinecraft
import com.andrew121410.mc.ccminecraftbot.packets.PacketHandler
import com.andrew121410.mc.ccminecraftbot.player.MovementManager
import com.andrew121410.mc.ccminecraftbot.world.Location.Companion.from
import com.github.steveice10.mc.protocol.packet.ingame.server.world.ServerSpawnPositionPacket

class OnServerSpawnPositionPacket : PacketHandler<ServerSpawnPositionPacket>() {
    override fun handle(packet: ServerSpawnPositionPacket, ccBotMinecraft: CCBotMinecraft) {
        val ccPlayer = ccBotMinecraft.player
        ccPlayer.currentLocation = from(packet.position)
        ccPlayer.spawnPoint = from(packet.position)
        ccPlayer.movementManager = MovementManager(ccPlayer)
        ccPlayer.movementManager!!.onJoin()
    }
}